### PR TITLE
kubectl command wrapper

### PR DESCRIPTION
**Pull request by Franr**
## What?
Add support to run `kubectl` commands from the leverage CLI.
It works by taking the `kubectl` config file from your home folder and passing the commands directly to the binary.
The authentication for AWS (required for EKS) is handled by `leverage` transparently (although is slow, so a topic open for discussion).

## Why?
To keep unifying flows into a single tool. **[More details here](https://github.com/binbashar/leverage/issues/126)**.

## References
New toolbox image required: https://github.com/binbashar/le-docker-leverage-toolbox/pull/43

## Open questions (Update: issues were created for all of them already!)

- **What version of `kubectl` do we want to support**? The binaries are kind of lightweight (~40MB) so we can have multiples on the  image and provide a command to do the switch.
- **Do we want automatic refreshing of the tokens like we do with the `terraform` commands**? That is the approach I took, but is proved to be slow (takes ~2 seconds every time you execute a command). For `terraform` is feasible but for `kubectl` looks like is not. I would like to take another approach.
- **Should we add a layer of commands on top of it**? Like a wrapper for get/use-context. We should analyze the complexity of it before adventuring into this direction, doesn't seems trivial.

## ToDo
- [x] unit tests
- [x] flow documentation (login, set/switch contexts, run commands)
- [x] migrate token generation/refresh into python?

### CODE REVIEW

**Comment by diego-ojeda-binbash** [comment]
@exequielrafaela same comment here about reviewers and PR state 🙏

**Comment by exequielrafaela** [comment]
> @exequielrafaela same comment here about reviewers and PR state 🙏

@diego-ojeda-binbash agreed as stated in the [related PR](https://github.com/binbashar/le-docker-leverage-toolbox/pull/43#issuecomment-1435244409) 🤝 ✅ 💪🏼

File: leverage/_utils.py [hunk 1]
```diff
@@ -50,3 +50,34 @@ def git(command):
     command = ["git"] + command if command[0] != "git" else command
 
     run(command, stdout=PIPE, stderr=PIPE, check=True)
+
+
+def chain_commands(commands: list, chain: str = " && ") -> str:
+    return f"bash -c \"{chain.join(commands)}\""
+
+
+class CustomEntryPoint:
```
Comment (angelofenoglio) [inline leverage/_utils.py:59]: Love the use of context managers, very elegant solution!

File: leverage/_utils.py [hunk 2]
```diff
@@ -50,3 +56,78 @@ def git(command):
...
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        container = args[0]  # this is the "self" of the method you are decorating; a LeverageContainer instance
+
+        if container.sso_enabled:
+            container._check_sso_token()
+            auth_method = container.TF_SSO_ENTRYPOINT
+        elif container.mfa_enabled:
+            auth_method = container.TF_MFA_ENTRYPOINT
+            # TODO: ask why this was necessary
```
Comment (Franr) [inline leverage/_utils.py:106]: @angelofenoglio @juanmatias maybe you can clarify me this guys?

**Review by angelofenoglio — APPROVED** [review]

**Pull request merged**